### PR TITLE
[terra-slide-group] Fix VoiceOver not announcing title of nested panel

### PR DESCRIPTION
**Pull request by smason0**
### Summary

**What was changed:**
Added VisuallyHiddenText to Slide component within SlideGroup to be conditionally visible to screen readers when the slide is nested. This text should be the aria-label of the slide (typically the title).

**Why it was changed:**
When using the VoiceOver screen reader, the aria-label of the initial slide in the SlideGroup was announced, but any nested slide was not announced upon being opened. This change allows the aria-label of nested slides to be announced to provide better context for screen reader users.

### Testing

This change was tested using:

- [ ] WDIO
- [x] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [x] Other (please describe below)
- [ ] No tests are needed

Tested using VoiceOver Utility version 10 (869.25) on MacOS Ventura 13.5 in Safari browser. Also tested with JAWS screen reader in Edge browser to ensure passive behavior.

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [x] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-8971 

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-slide-group/src/Slide.jsx [hunk 1]
```diff
@@ -29,6 +30,12 @@ const propTypes = {
    * The aria label for the Slide.
    */
   slideAriaLabel: PropTypes.string,
+
+  /**
+   * @private
```
Comment (kenk2) [inline packages/terra-slide-group/src/Slide.jsx:35]: Question: Why is this a private method?

**Review by kenk2 — COMMENTED** [review]
Just one question, otherwise seems good and I can approve it once answered.

File: packages/terra-slide-group/src/Slide.jsx (hunk 1 above)
Comment (smason0) [reply packages/terra-slide-group/src/Slide.jsx:35]: Good point. When I originally added this, I was under the impression that Slide would possibly be consumed externally, but it is only used as an internal component within SlideGroup. Therefore, I can remove this tag since the props for Slide are all essentially "private" anyway and never exposed to clients via the docs.

File: packages/terra-slide-group/tests/jest/SlideGroup.test.jsx [hunk 2]
```diff
@@ -14,11 +14,39 @@ const slideGroup = (
 
 it('should render a default SlideGroup', () => {
   const wrapper = shallow(slideGroup);
+
+  const cssTransitions = wrapper.find('CSSTransition');
+  expect(cssTransitions.at(0).key()).toBe('1');
+  expect(cssTransitions.at(0).prop('enter')).toBe(false);
+  expect(cssTransitions.at(0).prop('enter')).toBe(false);
```
Comment (kenk2) [inline packages/terra-slide-group/tests/jest/SlideGroup.test.jsx:21]: One last question: Why are these lines being double-repeated, I shouldn't expect a change here when I call the same `expect` shouldn't I?

File: packages/terra-slide-group/tests/jest/SlideGroup.test.jsx [hunk 3]
```diff
@@ -14,11 +14,39 @@ const slideGroup = (
...
+  expect(cssTransitions.at(2).find('Slide').prop('isNested')).toBe(true);
+
   expect(wrapper).toMatchSnapshot();
 });
 
 it('should render a SlideGroup with animation enabled', () => {
   const wrapper = shallow(React.cloneElement(slideGroup, { isAnimated: true }));
+
+  const cssTransitions = wrapper.find('CSSTransition');
+  expect(cssTransitions.at(0).prop('enter')).toBe(true);
```
Comment (kenk2) [inline packages/terra-slide-group/tests/jest/SlideGroup.test.jsx:43]: One last nit to add some whitespace between each `item`

**Review by kenk2 — COMMENTED** [review]
Just a few nitpicks, otherwise LGTM

File: packages/terra-slide-group/tests/jest/SlideGroup.test.jsx (hunk 2 above)
Comment (smason0) [reply packages/terra-slide-group/tests/jest/SlideGroup.test.jsx:21]: Oops, yeah the other ones should be the "exit" prop. Copy/paste error I will fix now. I will change the grouping also

**Review by kenk2 — APPROVED** [review]
LGTM!

**Review by trandrew1023 — APPROVED** [review]

**Review by eawww — APPROVED** [review]
VO announces name of nested disclosures.
JAWS announces name of nested disclosures and previous disclosures when nested ones are dismissed.
